### PR TITLE
📖  docs: Add installation guide with homebrew

### PR DESCRIPTION
**Pull request by ludusrusso**
**What this PR does / why we need it**:

Add installation guide with homebrew for macOS in book. Related to #2389 

Can be merged once Homebrew/homebrew-core/73610 is merged!

### CODE REVIEW

**Comment by ludusrusso** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please follow instructions at https://git.k8s.io/community/CLA.md#the-contributor-license-agreement to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment and we'll verify. Thanks.
> 
> * If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address.  Check your existing CLA data and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * If you signed the CLA as a corporation, please sign in with your organization's credentials at https://identity.linuxfoundation.org/projects/cncf to be authorized.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please log a ticket with the Linux Foundation Helpdesk: https://support.linuxfoundation.org/
> * Should you encounter any issues with the Linux Foundation Helpdesk, send a message to the backup e-mail support address at: [login-issues@jira.linuxfoundation.org](mailto:login-issues@jira.linuxfoundation.org)

Done

**Comment by ludusrusso** [comment]
I signed it

**Comment by ludusrusso** [comment]
I signed it

**Comment by davideimola** [comment]
What about to add a chapter on updating clusterctl with brew?

**Comment by fabriziopandini** [comment]
/ok-to-test
/hold 
for the discussion on #2389 
  
> What about to add a chapter on updating clusterctl with brew?

This is a good idea, but IMO this should not go in the quick start. what about having something under tasks? Also, If I'm not wrong, we are also still discussing how to get automatic upgrades of the formula...

**Comment by ludusrusso** [comment]
> /ok-to-test
> /hold
> for the discussion on #2389
> 
> > What about to add a chapter on updating clusterctl with brew?
> 
> This is a good idea, but IMO this should not go in the quick start. what about having something under tasks? Also, If I'm not wrong, we are also still discussing how to get automatic upgrades of the formula...

I agree with @fabriziopandini! I'll also put this in WIP until we decide the best name for the homebrew formula!

File: docs/book/src/user/quick-start.md [hunk 1]
```diff
@@ -125,6 +125,20 @@ Test to ensure the version you installed is up-to-date:
 ```
 clusterctl version
 ```
+
+##### Install clusterctl with homebrew on macOS
+
+Install the latest realease using homebrew:
```
Comment (stmcginnis) [inline docs/book/src/user/quick-start.md:131]: s/realease/release/

**Review by stmcginnis — COMMENTED** [review]
I wonder if we should get rid of the macOS curl instructions. Though no harm really having both options explained.

Looks good to me, but one small typo pointed out.

**Comment by stmcginnis** [comment]
Brew use on Linux still seems relatively low, but I could see it being useful.

Just another thought, but we could have a separate `brew` tab. Then have Linux, macOS, and Brew specific instructions, since there really isn't any OS-specific difference when using brew to install. Just throwing the idea out there.

**Comment by vincepri** [comment]
+1 on having a brew tab as an alternative method.

/ok-to-test

**Comment by ludusrusso** [comment]
Schermata 2021-03-23 alle 16 43 05

@vincepri @stmcginnis what do you think?

**Comment by stmcginnis** [comment]
Love it!

File: docs/book/src/user/quick-start.md [hunk 2]
```diff
@@ -125,6 +125,22 @@ Test to ensure the version you installed is up-to-date:
...
+{{#/tab }}
+{{#tab homebrew}}
+
+##### Install clusterctl with homebrew on macOS and linux
+
+Install the latest release using homebrew:
+
+```bash
+brew install clusterctl
+```
```
Comment (vincepri) [inline docs/book/src/user/quick-start.md:137]: One quick question on this:

- How are we going to handle upgrading within a release series? Like upgrade between v0.3.14 to v0.3.15 for example.
- How can we support multiple release series? Like 0.3 for now, and 0.4 later.

Comment (ludusrusso) [reply docs/book/src/user/quick-start.md:137]: Going to open a PR on #2389

Regarding multiple release, homebrew supports installation like `brew install kubectl@0.3` out of the box.

Comment (ludusrusso) [reply docs/book/src/user/quick-start.md:137]: Sorry wrong issue #4356 :D

**Comment by ludusrusso** [comment]
I think we are ready on this!

**Review by fabriziopandini — COMMENTED** [review]
/lgtm

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**